Align with the 2020-04-17 release of the algorithm documentation

bzg opened the pull request.

Commit:
Don't show intermediary urgence message

Fix https://github.com/etalab/covid19-orientation/issues/100

It works but this is a work in progress: some functions' signatures
need to be updated for emergency messages to be completely removed
from the codebase.

Commit:
Finish removing references to urgent message handling

Commit:
Enhance explanations for the postal code question

Fix https://github.com/etalab/covid19-orientation/issues/99

Commit:
Enhance UI message for height/weight question

Commit:
imc: Enhance messages

Commit:
Update min/max allowed values for height/weight

Commit:
Validate height/weight with a more explicit message

Partial fix to https://github.com/etalab/covid19-orientation/issues/94

Commit:
Explain what to do if the respondant does not know the postal code

Partial fix to https://github.com/etalab/covid19-orientation/issues/95

Commit:
Don't display final thanks message

Fix https://github.com/etalab/covid19-orientation/issues/98

Commit:
Don't ask the preliminary question about fever

Also don't make calculations depending on the deleted `fever`
variable and don't send `fever` to the API, only `feverAlgo`.

Commit:
Bump `algo_version` and `form_version` to "2020-04-16"

Anticipating a new release of the algorithm documentation at
https://github.com/Delegation-numerique-en-sante/covid19-algorithme-orientation

Commit:
Fix a conditional check to set the orientation

In "Tout patient sans fièvre avec un seul symptôme parmi toux,
douleurs, anosmie" (section 3.5 of the algorithm documentation),
it should be _only one_ symptom (XOR), not _at least one_ (OR).

Commit:
postal-code.js: Small rephrasing

Thanks to Julien Dudebout for the suggestion.

Commit:
Revert "postal-code.js: Small rephrasing"

This reverts commit b9f508b7e495eb8d85f084f9848d9336393afcf9.

Commit:
Implement last change from 2020-04-17 release of the algorithm

See https://github.com/Delegation-numerique-en-sante/covid19-algorithme-orientation/blob/master/ChangeLog.org#2020-04-17

Commit:
Update algo_version and form_version to refer to latest algo release

Commit:
Revert "Update algo_version and form_version to refer to latest algo release"

This reverts commit 16327ad4a11f8837d7bef5dc6613574ec8268ac5.

Commit:
Revert "Revert "Update algo_version and form_version to refer to latest algo release""

This reverts commit 5a0912bda5fe58cf64b6e277b6b5b62de588d6e5.

revolunet commented on components/respondent/imc.js:
```diff
@@ -41,20 +41,20 @@ function IMC({handleHeight, handleWeight}) {
+                  data-check-number data-check-number-min='25'
+                  data-check-number-max='400'
+                  data-check-number-error='Ce nombre ne semble pas être un poids valide en kilogrammes.'
+                  placeholder='Par ex. 70'
```
je mettrai input `type="number"` et les attributs `min` et `max`. 

https://developer.mozilla.org/fr/docs/Web/HTML/Element/Input/number

revolunet commented on components/respondent/imc.js:
```diff
@@ -41,20 +41,20 @@ function IMC({handleHeight, handleWeight}) {
           {height && weight && (
             <button className='mainbutton' type='submit'>
-              <span>Valider ces informations et continuer</span><i className='fas fa-check' aria-hidden='true' />
+              <span>Je mesure {height} cm et je pèse {weight} kilos.</span><i className='fas fa-check' aria-hidden='true' />
```
mettre un message particulier + bouton disabled si `height` ou `weight` sortent du range

revolunet commented:
Difficile de mon prononcer sur l'algo, il faudrait faire quelques TUs sur la fonction `chooseEnd`

Et il faudrait imho améliorer la validation pour l'IMC

bzg commented:
> Difficile de mon prononcer sur l'algo, il faudrait faire quelques TUs sur la fonction `chooseEnd`

J'imagine que c'est trop long à écrire maintenant.  Peut-être quelques tests manuels pour voir si on tombe sur la même chose que via le [démonstrateur](https://delegation-numerique-en-sante.github.io/covid19-algorithme-orientation/demonstrateur-bis.html) ?

> Et il faudrait imho améliorer la validation pour l'IMC

Est-ce que ce serait long d'implémenter [cette suggestion](https://github.com/etalab/covid19-orientation/issues/94#issuecomment-611486113)?

revolunet commented:
non ce ne serait pas long mais je n'aurai pas le temps de mon côté :/ je lance une bouteille à la mer pour @sixertoy au cas où

jdesboeufs reviewed (commented):
Je ne vois rien de choquant.

bzg commented:
> Je ne vois rien de choquant.

Ok, merci d'avoir regardé -  pour info je viens de discuter avec @sixertoy qui avance sur les tests unitaires.

sixertoy commented:
Hello,
Désolé pour le emoji suspect sur le comment (que je viens de delete du coup)
Je souhaitais notifier que je bosse dessus 
J'étais en mode distanciation "sociale de code", my bad je ferais mieux sur la communication :)

sixertoy commented on pages/index.js:
```diff
@@ -83,7 +83,9 @@ const chooseEnd = ({
-  } else if (cough || soreThroatAches || agueusiaAnosmia) {
+  } else if ((cough && !soreThroatAches && !agueusiaAnosmia) ||
+	     (!cough && soreThroatAches && !agueusiaAnosmia) ||
+	     (!cough && !soreThroatAches && !agueusiaAnosmia)) {
```
Remplacer
`(!cough && !soreThroatAches && !agueusiaAnosmia)`

par
`(!cough && !soreThroatAches && agueusiaAnosmia)`

la condition est un doublon du dernier use case
`IF NOT toux AND NOT douleurs AND NOT anosmie => FIN8`

Mais du coup plus besoin de faire le changement du IF non ? 🤔

bzg replied:
Corrigé dans 03e7544a03 - merci ! On aurait besoin d'un XOR, pas de `||`, car il faut un _seul_ des trois symptômes ici.

(Comme la condition précédente a aussi changé, il n'est effectivement plus logiquement nécessaire d'avoir un XOR ici... car le cas où il y a deux symptômes en même temps est traité dans cette condition précédente - mais l'algorithme a été validé tel quel par les médecins, je ne veux pas relancer une discussion sur l'économie logique de l'arbre - gardons cela.)

sixertoy commented on pages/index.js:
```diff
@@ -63,7 +63,7 @@ const chooseEnd = ({
+  } else if (feverAlgo || (!feverAlgo && (diarrhea || (cough && soreThroatAches) || (agueusiaAnosmia && soreThroatAches) || (cough && agueusiaAnosmia)))) {
     if (pronosticFactorsCount === 0) {
       if (minorSeverityFactorsCount === 0) {
         if (ageRange === 'inf_15' || ageRange === 'from_15_to_49') {
```
La condtion `ageRange === 'inf_15'` ne peut pas arriver
elle est bloquée par la première condition
`IF moins de 15 ans => FIN1`

Mais la seule condition `ageRange === 'from_15_to_49'` suffit pas à comprendre `moins de 50`
Je propose de créer des variables nommées pour remplacer les conditions

```javascript
const moinsDe50Ans = ageRange === 'inf_15' || ageRange === 'from_15_to_49'
const plusDe50Ans = ageRange === 'from_50_to_69' || ageRange === 'sup_70'

if (moinsDe50Ans) return 2
if (plusDe50Ans) return 3
```

bzg replied:
Merci ! J'ai corrigé dans 24e897c1. Pourquoi pas pour les valeurs intermédiaires: je t'ajoute en collaborateur sur le dépôt pour que tu puisses pousser la modification directement.

Commit:
pages/index.js: Fix code typo

Thanks to @sixertoy for catching this.

Commit:
pages/index.js: Remove useless check

Thanks to @sixertoy for catching this.

sixertoy committed:
✅ ajout des tests unitaires sur la function chooseEnd

- split de la function chooseEnd dans le dossier utils
- add jest as dependeny + add .nvmrc file
- add jest env to xo config

Commit:
Merge pull request #104 from sixertoy/wip

Ajout des tests unitaires sur la function chooseEnd

sixertoy committed:
(#105) 🎨 fix main button css when questions are required

sixertoy committed:
(#105) 🧁  replace in HTML <label for='' by htmlFor='' (react)

Commit:
Merge pull request #106 from sixertoy/wip

(#105) Incohérence dans l'affichage du bouton pour les pages de choix radio

sixertoy committed:
(#105) 🎨 boulette css, du code traine sans accolades

Commit:
Merge pull request #107 from sixertoy/wip

(#105) 🎨 boulette css, du code traine sans accolades

Commit:
Merge branch 'master' into wip

sixertoy committed:
(#99) ajout d'une checkbox pour le bypass du code postal user

Commit:
Merge pull request #109 from sixertoy/wip

(#99) ajout d'une checkbox pour le bypass du code postal user

sixertoy committed:
(#110) 👙 ajout des min max pour les champs IMC

Commit:
Merge pull request #112 from sixertoy/wip

(#110) 👙 ajout des min max pour les champs IMC

sixertoy committed:
(#113) ✅ ajout des TU sur la function computeIMC

Commit:
Merge pull request #114 from sixertoy/wip

(#113) ✅ ajout des TU sur la function computeIMC

Commit:
Revert "fix(fever): ensure feverAlgo is computed correctly"

This reverts commit 3c0f52e1961ed3341cf805b8ad4a942a88341b05.

Commit:
Revert "(#110) 👙 ajout des min max pour les champs IMC"

This reverts commit 6d458fd33c8f6a3696453f29c646ea32a153cc06.

Commit:
pages/index.js: set form_version to 2020-04-22

Commit:
README.md: Add instructions on how to test locally

Commit:
Merge branch 'master' into wip
